django_mongodb/lookups.py

```diff
@@ -0,0 +1,128 @@
+from django.core.exceptions import EmptyResultSet, FullResultSet
+from django.db.models.expressions import Col
+from django.db.models.fields.related_lookups import In, MultiColSource, RelatedIn
+from django.db.models.lookups import (
+    BuiltinLookup,
+    Contains,
+    Exact,
+    IntegerFieldOverflow,
+    IsNull,
+    StartsWith,
+)
+from django.db.models.sql.where import AND, XOR, WhereNode
+
+from ._helpers import process_lhs, process_rhs, safe_regex
+
+
+def where_node(self, compiler, connection):
+    if self.connector == AND:
+        full_needed, empty_needed = len(self.children), 1
+    else:
+        full_needed, empty_needed = 1, len(self.children)
+
+    if self.connector == AND:
+        operator = "$and"
+    elif self.connector == XOR:
+        raise NotImplementedError
+    else:
+        operator = "$or"
+
+    children_mql = []
+    for child in self.children:
+        try:
+            mql = child.as_mql(compiler, connection)
+        except EmptyResultSet:
+            empty_needed -= 1
+        except FullResultSet:
+            full_needed -= 1
+        else:
+            if mql:
+                children_mql.append(mql)
+            else:
+                full_needed -= 1
+
+        if empty_needed == 0:
+            raise (FullResultSet if self.negated else EmptyResultSet)
+        if full_needed == 0:
+            raise (EmptyResultSet if self.negated else FullResultSet)
+
+    if len(children_mql) == 1:
+        mql = children_mql[0]
+    elif len(children_mql) > 1:
+        mql = {operator: children_mql} if children_mql else {}
+    else:
+        mql = {}
+
+    if not mql:
+        raise FullResultSet
+
+    if self.negated and mql:
+        lhs, rhs = next(iter(mql.items()))
+        mql = {lhs: {"$not": rhs}}
+
+    return mql
+
+
+def exact(self, compiler, connection):
+    lhs_mql = process_lhs(self, compiler, connection)
+    value = process_rhs(self, compiler, connection)
+    if isinstance(self.lhs, Col):
+        lhs_mql = f"${lhs_mql}"
+    return {"$expr": {"$eq": [lhs_mql, value]}}
+
+
+def integer_field_overflow(self, compiler, connection):
+    rhs = self.rhs
```

Comment:
could all this be replaced with `self.process_rhs()`?

Reply:
Now, It is handled by the building_lookup. Removed.